### PR TITLE
[4.0] Move javascript to be inline

### PR DESCRIPTION
**Pull request by wilsonge**
Move javascript to enqueue through JDocument and convert from jquery to native js (@dgrammatiko i did something maybe good?)

To test: Ensure that when you have multiple 2FA plugins installed you can swap between them successfully

### CODE REVIEW

**Comment by brianteeman** [comment]
updated the title

**Review by dgrammatiko — APPROVED** [review]

File: components/com_users/tmpl/profile/edit.php [hunk 1]
```diff
@@ -17,6 +17,21 @@
 $lang = JFactory::getLanguage();
 $lang->load('plg_user_profile', JPATH_ADMINISTRATOR);
 
+JFactory::getDocument()->addScriptDeclaration("
+    Joomla.twoFactorMethodChange = function(e)
+    {
+        var selectedPane = 'com_users_twofactor_' + document.getElementById('jform_twofactor_method').value;
+
+        document.querySelector('#com_users_twofactor_forms_container>div').forEach(function(el) {
```
Comment (dgrammatiko) [inline components/com_users/tmpl/profile/edit.php]: [].slice.call(document....).forEach(...)

Ms edge and ie do not support iteration over nodelist so it needs to be converted to an array

**Review by dgrammatiko — CHANGES_REQUESTED** [review]

File: components/com_users/tmpl/profile/edit.php (hunk 1 above)
Comment (dgrammatiko) [reply components/com_users/tmpl/profile/edit.php]: And also it’s document.querySelectorAll() to get the nodelist, document.querySelector() returns the first element

**Comment by laoneo** [comment]
Should this not be moved to a script to remove inline at all? At least that's what we did in the ES6 repo too.

**Comment by wilsonge** [comment]
It's not a big deal either way. This was the minimum required for the CSP work we were doing at the JSST sprint.

**Comment by laoneo** [comment]
Still would be good to have it as its own script like we did with others. To keep consistency.

**Comment by wilsonge** [comment]
OK So I moved it to a standalone file. But looks like there's some sort of bug because the es6 files don't run through babel (I can't see anywhere where the es6 compiler is even called in `build.js`)

**Comment by wilsonge** [comment]
@dgrammatiko can you please ensure that we are actually running babel on the es6 files - bit of an issue if we are not!

**Comment by laoneo** [comment]
Commands are in the description of #20396.

**Comment by dgrammatiko** [comment]
@wilsonge https://github.com/joomla/joomla-cms/pull/20954 should do that.

There is a #todo for all the commands to become totally async, right now few are synchronous some are using promises and this will bite us at some point

**Comment by wilsonge** [comment]
I ran that command @laoneo - thats how i got my minified es6 file. it does not generate the es5 files... #20954 looks like the missing part tho!

**Comment by laoneo** [comment]
Ok, I assume you saw the two commands. Then we need the help of @dgrammatiko.

**Comment by wilsonge** [comment]
OK got there in the end.

**Comment by dgrammatiko** [comment]
Javascript is fun! Nobody said that in 2018... 😂

**Comment by laoneo** [comment]
Are the failing system tests related to this pr?

**Comment by brianteeman** [comment]
@laoneo I dont think so - I think that has been there since #20841 but I could be very wrong

**Comment by laoneo** [comment]
Drone was there green.

**Comment by laoneo** [comment]
I guess it was https://github.com/joomla/joomla-cms/pull/20919.

**Comment by laoneo** [comment]
So I'm going to merge this one.

**Pull request merged**